### ISSUE TITLE
`sort-comp` orders `static-variables` wrong.

### ISSUE THREAD

**Issue by forbid403** [yannickcr/eslint-plugin-react#3127]
Hello, I have a problem with `react/sort-comp` rule.

This is config file:
```json
"react/sort-comp": [
	2,
	{
		"order": [
			"static-variables",
			"static-methods",
			"type-annotations",
			"instance-variables",
			"lifecycle",
			"everything-else",
			"render"
		]
	}
```

And this is my code,

```tsx
class ClassName extends React.Component {
	static defaultProps = {};
	static parseDateString(date?: Date) {}
	state = {};
	...	
}
```

This code should not occur `sort-comp` error, but eslint says
`ESLint: defaultProps should be placed after parseDateString(react/sort-comp)`.

The `eslint-plugin-react` version is `7.26.1`

Thank you.

**Comment by ljharb** [comment]
This is expected. `defaultProps` is grouped under "lifecycle", not "static-variables".

**Comment by forbid403** [comment]
If I change `defaultProps` to anything else, still error occurs.
![스크린샷 2021-11-10 오후 4 05 56](https://user-images.githubusercontent.com/41617388/141066016-c543dc47-e1c3-4f82-8376-4b3ba99eca1f.png)

**Comment by ljharb** [comment]
In that case, it's clearly a bug :-)
